sprite.go

```diff
@@ -485,30 +485,22 @@ func (p *Sprite) OnTurning__1(onTurning func()) {
 	})
 }
 
-func (p *Sprite) Die() { // prototype sprite can't be destroyed, but can die
+func (p *Sprite) Die() {
 	aniName := p.getStateAnimName(StateDie)
 	p.SetDying()
 	if ani, ok := p.animations[aniName]; ok {
 		p.goAnimate(aniName, ani)
 	}
-	if p.isCloned_ {
-		p.doDestroy()
-	} else {
-		p.Hide()
-	}
-}
 
-func (p *Sprite) Destroy() { // delete this clone
```

Comment:
As proposed [here](https://github.com/goplus/spx/issues/314#issuecomment-2309491089), a public method named `DeleteThisClone` is expected to be offered:

> * Rename current `destroy` to `deleteThisClone` for compatibility with Scratch's [`Delete This Clone`](https://en.scratch-wiki.info/wiki/Delete_This_Clone_(block))